src/Compilers/Core/Portable/SourceGeneration/GeneratorDriverOptions.cs

```diff
@@ -3,21 +3,24 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
-using System.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
-    /// Options passed to a <see cref="GeneratorDriver"/> during creation
+    /// Options passed to a <see cref="GeneratorDriver"/> during creation.
     /// </summary>
     public readonly struct GeneratorDriverOptions
     {
         public readonly IncrementalGeneratorOutputKind DisabledOutputs;
 
         public readonly bool TrackIncrementalGeneratorSteps;
 
-        internal string? BaseDirectory { get; init; }
+        /// <summary>
+        /// Absolute path to directory that generated source file paths are rooted with.
+        /// Usually the project's output directory unless <see cref="CommandLineArguments.GeneratedFilesOutputDirectory"/> is specified.
+        /// </summary>
+        public string? BaseDirectory { get; }
```

Comment:
Consider mentioning in the doc comment when this can be `null`.

Reply:
agreed.  when is this null?

Reply:
It's null when null is passed to the constructor, or the constructor that doesn't have the parameter is used.